src/platform/packages/shared/kbn-discover-contextual-components/src/data_types/logs/components/summary_column/utils.tsx

```diff
@@ -57,60 +72,151 @@ export const getUnformattedResourceFields = (doc: LogDocument): ResourceFields =
+  const serviceName = getFieldValue(doc, SERVICE_NAME_FIELD);
+  const eventOutcome = getFieldValue(doc, EVENT_OUTCOME_FIELD);
+  const agentName = getFieldValue(doc, AGENT_NAME_FIELD);
+  const transactionName = getFieldValue(doc, TRANSACTION_NAME_FIELD);
+  const transactionDuration = getFieldValue(doc, TRANSACTION_DURATION_FIELD);
+  const spanName = getFieldValue(doc, SPAN_NAME_FIELD);
+  const spanDuration = getFieldValue(doc, SPAN_DURATION_FIELD);
+
+  return {
+    [SERVICE_NAME_FIELD]: serviceName,
+    [EVENT_OUTCOME_FIELD]: eventOutcome,
+    [AGENT_NAME_FIELD]: agentName,
+    [TRANSACTION_NAME_FIELD]: transactionName,
+    [TRANSACTION_DURATION_FIELD]: transactionDuration,
+    [SPAN_NAME_FIELD]: spanName,
+    [SPAN_DURATION_FIELD]: spanDuration,
+  };
+};
+
+const DurationIcon = () => {
+  const { euiTheme } = useEuiTheme();
+
+  return (
+    <EuiIcon
+      color="hollow"
+      type="clock"
+      size="m"
+      css={css`
+        margin-right: ${euiTheme.size.xs};
+      `}
+    />
+  );
+};
+
 /**
  * createResourceFields definitions
  */
 const AgentIcon = dynamic(() => import('@kbn/custom-icons/src/components/agent_icon'));
 
-const resourceCustomComponentsMap: Partial<
-  Record<keyof ResourceFields, React.ComponentType<FieldBadgeWithActionsProps>>
-> = {
-  [SERVICE_NAME_FIELD]: ServiceNameBadgeWithActions,
-};
-
 export interface ResourceFieldDescriptor {
   ResourceBadge: React.ComponentType<FieldBadgeWithActionsProps>;
   Icon?: () => JSX.Element;
-  name: keyof ResourceFields;
+  name: keyof ResourceFields | keyof TraceFields;
   value: string;
 }
 
-export const createResourceFields = (
+const getResourceBadgeComponent = (
+  name: keyof ResourceFields | keyof TraceFields,
+  core: CoreStart,
+  share?: SharePluginStart
+): React.ComponentType<FieldBadgeWithActionsProps> => {
+  switch (name) {
+    case SERVICE_NAME_FIELD:
+      return (props: FieldBadgeWithActionsProps) => (
+        <ServiceNameBadgeWithActions {...props} share={share} core={core} />
+      );
+    case EVENT_OUTCOME_FIELD:
+      return EventOutcomeBadgeWithActions;
+    default:
+      return FieldBadgeWithActions;
+  }
+};
+
+const getResourceBadgeIcon = (
+  name: keyof ResourceFields | keyof TraceFields,
+  fields: Readonly<ResourceFields> | Readonly<TraceFields>
+): (() => React.JSX.Element) | undefined => {
+  switch (name) {
+    case SERVICE_NAME_FIELD:
+      return () => {
+        const { euiTheme } = useEuiTheme();
+        return (
+          <AgentIcon
+            agentName={fields[AGENT_NAME_FIELD] as AgentName}
+            size="m"
+            css={css`
+              margin-right: ${euiTheme.size.xs};
+            `}
+          />
+        );
+      };
+    case EVENT_OUTCOME_FIELD:
+      return () => {
+        const { euiTheme } = useEuiTheme();
+
+        const value = (fields as Readonly<TraceFields>)[name as keyof TraceFields];
```

Comment:
You don't need to cast the name type.
```suggestion
        const value = (fields as Readonly<TraceFields>)[name];
```